src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,162 @@
+        userWithShortPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortPassword);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userWithNullPassword = new User();
+        userWithNullPassword.setLogin("validLogin");
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        User userWithEmptyPassword = new User();
+        userWithEmptyPassword.setLogin("validLogin");
+        userWithEmptyPassword.setPassword("");
+        userWithEmptyPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_minLengthLogin_ok() {
+        User userWithMinLengthLogin = new User();
+
+        String minLengthLogin = "MinLog";
+        userWithMinLengthLogin.setLogin(minLengthLogin);
+        userWithMinLengthLogin.setPassword("validPassword");
+        userWithMinLengthLogin.setAge(20);
+
+        registrationService.register(userWithMinLengthLogin);
+        assertEquals(userWithMinLengthLogin, storageDao.get(minLengthLogin));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User userWithEmptyLogin = new User();
+        userWithEmptyLogin.setLogin("");
+        userWithEmptyLogin.setPassword("validPassword");
+        userWithEmptyLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyLogin);
+        });
+    }
+
+    @Test
+    void register_ageIsZero_notOk() {
+        User userWithNullAge = new User();
+        userWithNullAge.setLogin("validLogin");
+        userWithNullAge.setPassword("validPassword");
+        userWithNullAge.setAge(0);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullAge);
+        });
+    }
+
+    @Test
+    void register_fewYears_notOk() {
+        User userWithFewYears = new User();
+        userWithFewYears.setLogin("validLogin");
+        userWithFewYears.setPassword("validPassword");
+        userWithFewYears.setAge(16);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithFewYears);
+        });
```

Comment:
The `register_fewYears_notOk` test case is not testing for edge cases. According to the checklist, it would be better to test for an age of 17 (which is the edge case just before the minimum valid age of 18).